### PR TITLE
DBP-686-clean-helm-chart

### PR DESCRIPTION
**Pull request by aimee-889**
# Description
- changed Helm Chart so that the Value structure is more conventional 
- tried to extract as many Envs that were taken from secrets be to taken from a configmap, the one thw the Vars in it $Host/$Name does not like to get transferred to the configmap, the Container will fail 
- I set the database.name here per defualt to spshdeveloper. In spsh-app-release this will get overwritten. But I would like it most if we also take the db name as db user name (this should be done outside this ticket)
- I took some content from the bitnami commin chart. But i did not add the common chart as dependency but directly added that to the templates file. Could maybe change it to use the Common Helm Chrat from Bitnami, open to comments. 

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: .github/workflows/image-and-helm-publish-check-deploy-on-push-scheduled.yml [hunk 1]
```diff
@@ -112,7 +112,7 @@ jobs:
       - create_branch_identifier
       - release_helm
       - build_image_on_push
-    uses: dBildungsplattform/spsh-app-deploy/.github/workflows/deploy.yml@2
+    uses: dBildungsplattform/spsh-app-deploy/.github/workflows/deploy.yml@DBP-686-clean-helm-chart
```
Comment (aimee-889) [inline .github/workflows/image-and-helm-publish-check-deploy-on-push-scheduled.yml:115]: change back before merge

File: charts/dbildungs-iam-keycloak/templates/deployment.yaml [hunk 2]
```diff
@@ -1,145 +1,85 @@
...
-          {{- end }}
-
-          args:            
-            - "--cache"
-            - "local"
-            - "--hostname"
-            - "{{ .Values.keycloakHostname }}" 
-            - "--import-realm"
+          args:
+              - "--cache"
```
Comment (simoncolincap) [inline charts/dbildungs-iam-keycloak/templates/deployment.yaml:30]: The arguments are indented too much

File: charts/dbildungs-iam-keycloak/templates/secret.yaml [hunk 3]
```diff
@@ -2,20 +2,13 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  {{- if .Values.auth.existingSecret }}
-  name: {{ .Values.auth.existingSecret }}
-  {{- end }}
-  {{- if not .Values.auth.existingSecret }}
   name: {{ .Values.auth.name }}
```
Comment (simoncolincap) [inline charts/dbildungs-iam-keycloak/templates/secret.yaml:5]: I think auth.name should be called auth.secretName for better clarity

**Review by simoncolincap — APPROVED** [review]

**Review by dimapin — APPROVED** [review]

File: charts/dbildungs-iam-keycloak/values.yaml [hunk 4]
```diff
@@ -57,49 +50,49 @@ containerSecurityContext:
...
     port: web
 
 ingress:
   ingressClassName: nginx
+  path: /
   pathType: Prefix
-  
-  httpRelativePath: "/"
-  portnumber: 80
+  annotations:
```
Comment (aimee-889) [inline charts/dbildungs-iam-keycloak/values.yaml:85]: these values should in a future PR set to {} and the concrete vlaues should be given to DEV via spsh-app-deploy  (ans spsh-app-release).

**Pull request merged**